rllib/examples/env/pettingzoo_chess.py

```diff
@@ -0,0 +1,193 @@
+from pettingzoo import AECEnv
+from pettingzoo.classic.chess.chess import raw_env as chess_v5
+import copy
+from ray.rllib.env.multi_agent_env import MultiAgentEnv
+from typing import Dict, Any
+import chess as ch
+import numpy as np
+
+
+class MultiAgentChess(MultiAgentEnv):
+    """An interface to the PettingZoo MARL environment library.
+    See: https://github.com/Farama-Foundation/PettingZoo
+    Inherits from MultiAgentEnv and exposes a given AEC
+    (actor-environment-cycle) game from the PettingZoo project via the
+    MultiAgentEnv public API.
+    Note that the wrapper has some important limitations:
+    1. All agents have the same action_spaces and observation_spaces.
+       Note: If, within your aec game, agents do not have homogeneous action /
+       observation spaces, apply SuperSuit wrappers
+       to apply padding functionality: https://github.com/Farama-Foundation/
+       SuperSuit#built-in-multi-agent-only-functions
+    2. Environments are positive sum games (-> Agents are expected to cooperate
+       to maximize reward). This isn't a hard restriction, it just that
+       standard algorithms aren't expected to work well in highly competitive
+       games.
+    Examples:
+        >>> from pettingzoo.butterfly import prison_v3
+        >>> from ray.rllib.env.wrappers.pettingzoo_env import PettingZooEnv
+        >>> env = PettingZooEnv(prison_v3.env())
+        >>> obs = env.reset()
+        >>> print(obs)
+        # only returns the observation for the agent which should be stepping
+        {
+            'prisoner_0': array([[[0, 0, 0],
+                [0, 0, 0],
+                [0, 0, 0],
+                ...,
+                [0, 0, 0],
+                [0, 0, 0],
+                [0, 0, 0]]], dtype=uint8)
+        }
+        >>> obs, rewards, dones, infos = env.step({
+        ...                 "prisoner_0": 1
+        ...             })
+        # only returns the observation, reward, info, etc, for
+        # the agent who's turn is next.
+        >>> print(obs)
+        {
+            'prisoner_1': array([[[0, 0, 0],
+                [0, 0, 0],
+                [0, 0, 0],
+                ...,
+                [0, 0, 0],
+                [0, 0, 0],
+                [0, 0, 0]]], dtype=uint8)
+        }
+        >>> print(rewards)
+        {
+            'prisoner_1': 0
+        }
+        >>> print(dones)
+        {
+            'prisoner_1': False, '__all__': False
+        }
+        >>> print(infos)
+        {
+            'prisoner_1': {'map_tuple': (1, 0)}
+        }
+    """
+
+    def __init__(
+        self,
+        config: Dict[Any, Any] = None,
+        env: AECEnv = chess_v5(),
```

Comment:
Can you make sure your constructor does not need the env object to be passed in? This will create issue for serializing the environment. One thing you could do is have config include a base env key in it and if None, default to chess_v5() inside the constructor. Thanks.